### ISSUE TITLE
[Bug]: NodeId doesn't update within replica after query node offine

### ISSUE THREAD

**Issue by jaime0815** [milvus-io/milvus#17581]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

queryNode offline at  10:19:10
```
[2022/06/15 10:19:10.084 +00:00] [INFO] [cluster.go:645] [\"stopNode: queryNode offline\"] [nodeID=30]
```

but load balance still continuously tries to get an offline querynode info at 12:42:26

```
{"log":"[2022/06/15 12:42:26.597 +00:00] [WARN] [query_coord.go:604] [\"loadBalanceSegmentLoop: get node info from QueryNode failed\"] [nodeID=30] [collection=433922258571100161] [replica=433921805915521034] [error=\"getNodeInfo: queryNode is offline\"]\n","stream":"stdout","time":"2022-06-15T12:42:26.59746057Z"}
{"log":"[2022/06/15 12:41:26.597 +00:00] [WARN] [query_coord.go:604] [\"loadBalanceSegmentLoop: get node info from QueryNode failed\"] [nodeID=30] [collection=433922258571100161] [replica=433921805915521034] [error=\"getNodeInfo: queryNode is offline\"]\n","stream":"stdout","time":"2022-06-15T12:41:26.597946728Z"}
{"log":"[2022/06/15 12:40:26.597 +00:00] [WARN] [query_coord.go:604] [\"loadBalanceSegmentLoop: get node info from QueryNode failed\"] [nodeID=30] [collection=433922258571100161] [replica=433921805915521034] [error=\"getNodeInfo: queryNode is offline\"]\n","stream":"stdout","time":"2022-06-15T12:40:26.59784853Z"}
{"log":"[2022/06/15 12:39:26.597 +00:00] [WARN] [query_coord.go:604] [\"loadBalanceSegmentLoop: get node info from QueryNode failed\"] [nodeID=30] [collection=433922258571100161] [replica=433921805915521034] [error=\"getNodeInfo: queryNode is offline\"]\n","stream":"stdout","time":"2022-06-15T12:39:26.597864947Z"}
{"log":"[2022/06/15 12:38:26.597 +00:00] [WARN] [query_coord.go:604] [\"loadBalanceSegmentLoop: get node info from QueryNode failed\"] [nodeID=30] [collection=433922258571100161] [replica=433921805915521034] [error=\"getNodeInfo: queryNode is offline\"]\n","stream":"stdout","time":"2022-06-15T12:38:26.597464047Z"}
{"log":"[2022/06/15 12:37:26.597 +00:00] [WARN] [query_coord.go:604] [\"loadBalanceSegmentLoop: get node info from QueryNode failed\"] [nodeID=30] [collection=433922258571100161] [replica=433921805915521034] [error=\"getNodeInfo: queryNode is offline\"]\n","stream":"stdout","time":"2022-06-15T12:37:26.597160638Z"}
```

**Comment by jaime0815** [comment]
{"log":"[2022/06/15 12:42:16.754 +00:00] [WARN] [impl.go:1188] [\"shard leader is unavailable\"] [collectionID=433922258571100161] [replicaID=433921805915521034] [DmChannel=in01-d22cf04e350d004-rootcoord-dml_1_433922258571100161v1] [shardLeaderID=30] []\n","stream":"stdout","time":"2022-06-15T12:42:16.754355592Z"}
 {"log":"[2022/06/15 12:42:16.754 +00:00] [WARN] [impl.go:1188] [\"shard leader is unavailable\"] [collectionID=433922258571100161] [replicaID=433921805915521034] [DmChannel=in01-d22cf04e350d004-rootcoord-dml_0_433922258571100161v0] [shardLeaderID=29] []\n","stream":"stdout","time":"2022-06-15T12:42:16.754353675Z"}

**Comment by jaime0815** [comment]
load segment task finishes at 10:18:09 on NodeID 30, then it offline at 10:19:10, but can not find any log about balance

```
{"log":"[2022/06/15 10:18:09.110 +00:00] [INFO] [impl.go:241] [\"loadSegmentsTask WaitToFinish done\"] [collectionID=433922258571100161] [segmentIDs=\"[433923392186613763,433922616921948162,433922797342818307,433922855591215107,433923509627387907]\"] [nodeID=30]\n","stream":"stdout","time":"2022-06-15T10:18:09.110996283Z"}

``

**Comment by jaime0815** [comment]
fix #17560
